oonidata/transforms/nettests/measurement_transformer.py

```diff
@@ -303,27 +303,47 @@ def find_tls_handshake_network_events(
     matched_event_windows = []
 
     current_event_window = []
+    all_event_windows_dict = {}
     for idx, ne in enumerate(network_events):
-        if ne.operation == "connect":
-            current_event_window = []
-        current_event_window.append(ne)
-        if ne.operation == "tls_handshake_done":
-            # We identify the network_event for the given TLS handshake based on the
-            # fact that the timestamp on tls_handshake_done event is the same as the
-            # tls_handshake time.
-            # In case of duplicates we also look for the index of the tls
-            # handshake inside of the list of event windows.
-            if ne.t == tls_handshake.t:
-                matched_event_windows.append(len(all_event_windows))
-            current_event_window += network_events_until_connect(network_events[idx:])
-            all_event_windows.append(current_event_window)
+        if ne.transaction_id == None:
+            if ne.operation == "connect":
+                current_event_window = []
+            current_event_window.append(ne)
+            if ne.operation == "tls_handshake_done":
+                # We identify the network_event for the given TLS handshake based on the
+                # fact that the timestamp on tls_handshake_done event is the same as the
+                # tls_handshake time.
+                # In case of duplicates we also look for the index of the tls
+                # handshake inside of the list of event windows.
+                if ne.t == tls_handshake.t:
+                    matched_event_windows.append(len(all_event_windows))
+                current_event_window += network_events_until_connect(network_events[idx:])
+                all_event_windows.append(current_event_window)
+        else:
+            transaction_id = ne.transaction_id
```

Comment:
Maybe it's even more robust to first do detection of the network_event data format capabilities and only after do this network event parsing and splitting logic.

That way we can handle early a scenario of, for example, some of the network_events not having a `transaction_id` set due to a bug (would it actually be a bug?) and fail early with a meaningful message, as opposed to having the logic in here handle different network event entries differently.